### DISCUSSION TITLE
is there a go code version for the same chat feature?

### DISCUSSION THREAD

**Discussion by gosgos11** [SomajitDey/ipfs-chat#22] [Q&A]
In a go application want to add same feature, is there a go code version for the same chat function? or other code? thank you.

**Comment by SomajitDey** [comment]
@gosgos11 Nothing in Go that I know of that's an exact mirror of this project.

If you are interested in just a messenger written in Go that uses IPFS or Libp2p you might take a look at [Berty](https://berty.tech/docs/berty/#main-items-in-the-repo), [Peerchat](https://github.com/manishmeganathan/peerchat) and [monk-seal](https://github.com/jamesthesken/monk-seal).

Many IPFS or Libp2p-based messengers actually use a JS-IPFS-node, interacting with a Go-IPFS node if needed with websockets.

**Reply by gosgos11** [reply]
> @gosgos11 Nothing in Go that I know of that's an exact mirror of this project.
> 
> If you are interested in just a messenger written in Go that uses IPFS or Libp2p you might take a look at [Berty](https://berty.tech/docs/berty/#main-items-in-the-repo), [Peerchat](https://github.com/manishmeganathan/peerchat) and [monk-seal](https://github.com/jamesthesken/monk-seal).
> 
> Many IPFS or Libp2p-based messengers actually use a JS-IPFS-node, interacting with a Go-IPFS node if needed with websockets.

Thank you for the reply, I will try the code.
